### PR TITLE
James-xxxx: FluentBit documentation

### PR DESCRIPTION
**Pull request by nvduc91**

### CODE REVIEW

File: docs/modules/servers/pages/distributed/operate/logging.adoc [hunk 1]
```diff
@@ -14,96 +14,77 @@ Cassandra timeouts). In such case we advise you to conduct a close
...
 == Logging configuration
 
-Distributed James uses link:http://logback.qos.ch/[logback] as a logging library.
+Distributed James uses link:http://logback.qos.ch/[logback] as a logging library
+and link:https://docs.fluentbit.io/[FluentBit] as centralize logging.
 
 Information about logback configuration can be found
 link:http://logback.qos.ch/manual/configuration.html[here].
 
 == Structured logging
```
Comment (chibenwa) [inline docs/modules/servers/pages/distributed/operate/logging.adoc:31]: We should have the following structure:

```
== Structured logging

=== Pushing logs to ElasticSearch

[... What was done before ]

=== Using fluentBit as a log forwarder

[ The new stuff ]

```
```

File: docs/modules/servers/pages/distributed/operate/logging.adoc [hunk 2]
```diff
@@ -14,96 +14,77 @@ Cassandra timeouts). In such case we advise you to conduct a close
...
-        </logger>
+        <root level="WARN">
+                <appender-ref ref="LOG_FILE" />
+        </root>
 
         <logger name="org.apache.james" level="INFO" />
+        <logger name="com.datastax.driver.core.QueryLogger.SLOW" level="DEBUG" />
 
 </configuration>
 ....
```
Comment (chibenwa) [inline docs/modules/servers/pages/distributed/operate/logging.adoc:90]: Please also share the fluentBit configuration you used

File: docs/modules/servers/pages/distributed/operate/logging.adoc [hunk 3]
```diff
@@ -38,72 +40,187 @@ link:https://www.elastic.co/elasticsearch[ElasticSearch].
 Here is a sample `conf/logback.xml` configuration file for logback with the following
 pre-requisites:
 
-* Logging both in an unstructured fashion on the console and in a structure fashion in
-ElasticSearch
+* Logging both in an unstructured fashion on the console and in a structure fashion in ElasticSearch
 * Logging ElasticSearch Log appender logs in the console
```
Comment (chibenwa) [inline docs/modules/servers/pages/distributed/operate/logging.adoc:44]: We no longer dispay the sample logback.xml file for ElasticSearch exporter?

File: docs/modules/servers/pages/distributed/operate/logging.adoc [hunk 4]
```diff
@@ -38,72 +40,187 @@ link:https://www.elastic.co/elasticsearch[ElasticSearch].
...
 
         <contextListener class="ch.qos.logback.classic.jul.LevelChangePropagator">
                 <resetJUL>true</resetJUL>
         </contextListener>
 
-        <appender name="CONSOLE" class="ch.qos.logback.core.ConsoleAppender">
-                <encoder>
-                        <pattern>%d{yyyy.MM.dd HH:mm:ss.SSS} %highlight([%-5level]) %logger{15} - %msg%n%rEx</pattern>
-                        <immediateFlush>false</immediateFlush>
+        <appender name="LOG_FILE" class="ch.qos.logback.core.rolling.RollingFileAppender">
```
Comment (chibenwa) [inline docs/modules/servers/pages/distributed/operate/logging.adoc:62]: Can we do that with the Console Appender instead ?

**Comment by chibenwa** [comment]
https://issues.apache.org/jira/browse/JAMES-3399

File: docs/modules/servers/pages/distributed/operate/logging.adoc [hunk 5]
```diff
@@ -38,72 +40,187 @@ link:https://www.elastic.co/elasticsearch[ElasticSearch].
 Here is a sample `conf/logback.xml` configuration file for logback with the following
 pre-requisites:
 
-* Logging both in an unstructured fashion on the console and in a structure fashion in
-ElasticSearch
+* Logging both in an unstructured fashion on the console and in a structure fashion in ElasticSearch
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/operate/logging.adoc:43]:
```suggestion
* Logging both in an unstructured fashion on the console and in a structured fashion in ElasticSearch
```

File: docs/modules/servers/pages/distributed/operate/logging.adoc [hunk 6]
```diff
@@ -38,72 +40,187 @@ link:https://www.elastic.co/elasticsearch[ElasticSearch].
...
 * Logging ElasticSearch Log appender logs in the console
 
+=== Using fluentBit as a log forwarder
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/operate/logging.adoc:46]:
```suggestion
=== Using FluentBit as a log forwarder
```

File: docs/modules/servers/pages/distributed/operate/logging.adoc [hunk 7]
```diff
@@ -38,72 +40,187 @@ link:https://www.elastic.co/elasticsearch[ElasticSearch].
...
+
+Distributed Server leverages the use of MDC in order to achieve structured logging, and better add context to the logged information. We furthermore ship json log in to file with RollingFileAppender on the classpath to easily allow FluentBit direct tail the log file.
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/operate/logging.adoc:48]:
```suggestion
Distributed Server leverages the use of MDC in order to achieve structured logging, and better add context to the logged information. We furthermore ship json logs to file with RollingFileAppender on the classpath to easily allow FluentBit to directly tail the log file.
```

File: docs/modules/servers/pages/distributed/operate/logging.adoc [hunk 8]
```diff
@@ -38,72 +40,187 @@ link:https://www.elastic.co/elasticsearch[ElasticSearch].
...
+Here is a sample conf/logback.xml configuration file for logback with the following pre-requisites:
+
+Logging in a structure json fashion and write to file for centralize logging.
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/operate/logging.adoc:51]:
```suggestion
Logging in a structured json fashion and write to file for centralizing logging.
```

File: docs/modules/servers/pages/distributed/operate/logging.adoc [hunk 9]
```diff
@@ -38,72 +40,187 @@ link:https://www.elastic.co/elasticsearch[ElasticSearch].
...
+Centralize logging third party like FluentBit can tail from logging’s file then filter/process and put in to ElastichSearch
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/operate/logging.adoc:52]:
```suggestion
To centralize logging, a third party software like FluentBit can tail from logging’s file then filter/process and put in to ElastichSearch
```

File: docs/modules/servers/pages/distributed/operate/logging.adoc [hunk 10]
```diff
@@ -38,72 +40,187 @@ link:https://www.elastic.co/elasticsearch[ElasticSearch].
...
+        </root>
 
         <logger name="org.apache.james" level="INFO" />
+        <logger name="com.datastax.driver.core.QueryLogger.SLOW" level="DEBUG" />
 
 </configuration>
 ....
+
+First you need to create a logs folder to mount the `logs` folder of James outside container,
+then mount that folder to FluentBit. In case you need to use `sh command` in fluent-bit container please use image: `fluent/fluent-bit:1.5.7-debug`
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/operate/logging.adoc:102]: First you need to create a `logs` folder, then mount it to James container and to FluentBit.

File: docs/modules/servers/pages/distributed/operate/logging.adoc [hunk 11]
```diff
@@ -38,72 +40,187 @@ link:https://www.elastic.co/elasticsearch[ElasticSearch].
...
+    environment:
+      ELASTICSEARCH_HOSTS: http://elasticsearch:9200
+    ports:
+      - "5601:5601"
+    depends_on:
+      - elasticsearch
+....
+
+FluentBit config as:
+the `Host elasticsearch` it point to `elasticsearch` service in docker-compose file.
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/operate/logging.adoc:184]:
```suggestion
the `Host elasticsearch` pointing to `elasticsearch` service in docker-compose file.
```

File: docs/modules/servers/pages/distributed/operate/logging.adoc [hunk 12]
```diff
@@ -38,72 +40,187 @@ link:https://www.elastic.co/elasticsearch[ElasticSearch].
...
+    Match *
+    Host elasticsearch
+    Port 9200
+    Index fluentbit
+    Logstash_Format On
+    Logstash_Prefix fluentbit-james
+    Type docker
+....
+
+FulentBit Parser config:
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/operate/logging.adoc:215]:
```suggestion
FluentBit Parser config:
```

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**